### PR TITLE
turn off dynamic shape for torch.compile

### PR DESCRIPTION
**Pull request by wanchaol**
as titled. This could make 1-D and 2-D works with the lastest main build. thanks @bdhirsh for all the fixes!

We should figure out why dynamic shape gets turned on as a follow up

### CODE REVIEW

File: train_configs/llama3_8b.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ save_tb_folder = "tb"
 [model]
 name = "llama3"
 flavor = "8B"
-norm_type = "fused_rmsnorm"  # [layernorm / np_layernorm / rmsnorm / fused_rmsnorm]
+norm_type = "rmsnorm"  # [layernorm / np_layernorm / rmsnorm / fused_rmsnorm]
```
Comment (awgu) [inline train_configs/llama3_8b.toml:21]: Is this change intentional?

**Review by awgu — APPROVED** [review]

File: train_configs/llama3_8b.toml (hunk 1 above)
Comment (wanchaol) [reply train_configs/llama3_8b.toml:21]: yeah I feel it's a bit painful to switch between `rmsnorm` and `fused_rmsnorm` locally when switching between 1-D and 2-D sharding, so I changed `rmsnorm` to default, we can always default to `fused_rmsnorm` once @tianyu-l changes the fused_rmsnorm registration https://github.com/pytorch/torchtitan/pull/296.

Let me know if you think we should default to `fused_rmsnorm` for 8B

Comment (awgu) [reply train_configs/llama3_8b.toml:21]: Keeping it as `rmsnorm` sounds good to me!
